sentry-rails/examples/rails-6.0/app/controllers/welcome_controller.rb

```diff
@@ -5,6 +5,18 @@ def index
     1 / 0
   end
 
+  def connect_trace
+    transaction = Sentry.get_current_scope.get_transaction
+    # see the sinatra example under the `sentry-ruby` folder
+    uri = URI("http://localhost:4567/connect_trace")
+    http = Net::HTTP.new(uri.host, uri.port)
+    request = Net::HTTP::Get.new(uri.request_uri)
+    request["SENTRY_TRACE"] = transaction.to_sentry_trace
```

Comment:
@st0012 if this line is required, then it explains the current difficulty in https://github.com/getsentry/sentry-ruby/issues/1411 as this would be a required manual step for trace propagation.

We try to make HTTP clients automatically propagate traces with auto instrumentation, perhaps that's missing in Ruby?

Reply:
I'm not aware of this requirement in the backend SDK 🤔  can you point me the implementation in `sentry-python` and `sentry-go`?

Reply:
Python: https://github.com/getsentry/sentry-python/blob/51987c57157102bbd32e1e7b084c26f4dc475d86/sentry_sdk/integrations/stdlib.py#L57-L112

JS Node: https://github.com/getsentry/sentry-javascript/blob/f7fc73313cb1f602dfbde856e116bb74f39dde81/packages/node/src/integrations/http.ts#L109-L123

JS Browser: https://github.com/getsentry/sentry-javascript/blob/13eea94a4da4334dd79f829a2657c03dbb50f780/packages/tracing/src/browser/request.ts#L87-L264

Go doesn't have this yet (know limitation), and when we do it won't be like Python and Ruby because we can't monkey-patch libraries.

But the idea is that we automatically create spans for outgoing requests and propagate the trace header `sentry-trace` with information on the current span (the one that was just created for the request).

Reply:
@rhcarvalho I see, thanks. I'll merge this PR now and then implemented the propagation. after that's done I'll update the example again.